pylon-core/src/main/kotlin/io/github/pylonmc/pylon/core/item/research/Research.kt

```diff
@@ -0,0 +1,201 @@
+package io.github.pylonmc.pylon.core.item.research
+
+import com.github.shynixn.mccoroutine.bukkit.launch
+import com.github.shynixn.mccoroutine.bukkit.ticks
+import io.github.pylonmc.pylon.core.config.PylonConfig
+import io.github.pylonmc.pylon.core.datatypes.PylonSerializers
+import io.github.pylonmc.pylon.core.i18n.PylonArgument
+import io.github.pylonmc.pylon.core.item.PylonItem
+import io.github.pylonmc.pylon.core.item.PylonItemSchema
+import io.github.pylonmc.pylon.core.item.research.Research.Companion.canUse
+import io.github.pylonmc.pylon.core.pluginInstance
+import io.github.pylonmc.pylon.core.recipe.RecipeTypes
+import io.github.pylonmc.pylon.core.registry.PylonRegistry
+import io.github.pylonmc.pylon.core.util.persistentData
+import io.github.pylonmc.pylon.core.util.pylonKey
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.delay
+import net.kyori.adventure.text.Component
+import net.kyori.adventure.text.event.ClickEvent
+import net.kyori.adventure.text.event.HoverEvent
+import org.bukkit.GameMode
+import org.bukkit.Keyed
+import org.bukkit.NamespacedKey
+import org.bukkit.entity.Player
+import org.bukkit.event.EventHandler
+import org.bukkit.event.EventPriority
+import org.bukkit.event.Listener
+import org.bukkit.event.entity.EntityPickupItemEvent
+import org.bukkit.event.player.PlayerJoinEvent
+import org.bukkit.event.player.PlayerQuitEvent
+import java.util.UUID
+
+data class Research(
+    private val key: NamespacedKey,
+    val name: Component,
+    val cost: Long,
+    val unlocks: Set<NamespacedKey>
+) : Keyed {
+
+    constructor(key: NamespacedKey, cost: Long, vararg unlocks: PylonItemSchema) : this(
```

Comment:
```suggestion
    @SafeVarargs
    constructor(key: NamespacedKey, cost: Long, vararg unlocks: PylonItemSchema) : this(
```
just to prevent any warnings when its used in java

Reply:
`SafeVarargs` is only needed for generics